src/test/java/com/google/sps/servlets/CasesDataServletTest.java

```diff
@@ -36,25 +38,32 @@
 public final class CasesDataServletTest {
   @Mock private HttpServletRequest request;
   @Mock private HttpServletResponse response;
+  @Mock private ServletOutputStream mockOutput;
   private CasesDataServlet servlet;
 
   @Test
   public void servletBehavesCorrectly() throws IOException {
     MockitoAnnotations.initMocks(this);
     servlet = new CasesDataServlet();
-    when(response.getContentType()).thenReturn(servlet.CTYPE);
-    when(response.getCharacterEncoding()).thenReturn(servlet.ENCODING);
+    servlet.init();
+    String reportsJson = servlet.getReportsJson();
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
     when(response.getStatus()).thenReturn(HttpURLConnection.HTTP_OK);
     when(request.getMethod()).thenReturn("get");
     when(request.getProtocol()).thenReturn("http");
+    when(response.getWriter()).thenReturn(writer);
     servlet.doGet(request, response);
 
-    Assert.assertEquals("application/json", response.getContentType());
-    Assert.assertEquals("UTF-8", response.getCharacterEncoding());
-    Assert.assertEquals("/WEB-INF/cases.csv", servlet.STATS);
+    verify(response).setContentType("application/json");
+    verify(response).setCharacterEncoding("UTF-8");
+    verify(response).getWriter();
     Assert.assertEquals(Collections.<String>emptyList(), response.getHeaderNames());
     Assert.assertEquals(200, response.getStatus());
-    Assert.assertEquals("get", request.getMethod());
-    Assert.assertEquals("http", request.getProtocol());
+    Assert.assertTrue(reportsJson.contains("\"lat\":33.034"));
+    Assert.assertTrue(reportsJson.contains("\"lng\":-116.736"));
+    Assert.assertTrue(reportsJson.contains("\"lat\":41.591"));
+    Assert.assertTrue(reportsJson.contains("\"lng\":1.520"));
+    Assert.assertTrue(reportsJson.contains("\"active\""));
```

Comment:
Can you verify the data in the response? Sorry I didn't make it clearer before. It seems weird to test the data file specifically from the injected getReportsJson() method. What we really want to test is: whether the response has the data we need.

Feel free to remove the getReportsJson() and verify these in the response.

Reply:
If I verify the data in the response, I have to check that the writer printed the entire output, which would not work in our case because that output would change and it is long. This way, I check that the writer is called, and that the reportsJson has some correct data that we know will not change. If the writer is called and the reportsJson is correct, then the data in the response has to be correct